gradio/upload.py

```diff
@@ -0,0 +1,173 @@
+import os
+import random
+import huggingface_hub
+import json
+
+
+def get_existing_configuration(readme_file):
+    with open(readme_file, "r") as f:
+        readme = f.read()
+    if not readme.strip().startswith("---"):
+        return None
+    readme = readme.strip()
+    if "---" not in readme[3:]:
+        return None
+    readme = readme.split("---")[1].strip()
+    config = {}
+    for line in readme.split("\n"):
+        if ":" not in line:
+            continue
+        key, value = line.split(":", 1)
+        config[key.strip()] = value.strip()
```

Comment:
`str.partition` is faster:
```suggestion
        key, sep, value = line.partition(":")
        if sep:
            config[key.strip()] = value.strip()
```